js/views/modals/purchase/Purchase.js

```diff
@@ -140,19 +148,37 @@ export default class extends BaseModal {
   }
 
   purchaseListing() {
+    // set the address
+    this.order.addAddress(this.shipping.selectedAddress);
     // set the moderator
     this.order.set('moderator', this.moderators.selectedIDs[0]);
+    // set the shipping option
+    const oShipping = this.order.get('items').at(0).get('shipping');
+    const sOption = this.shipping.shippingOptions.selectedOption;
+    oShipping.set({ name: sOption.name, service: sOption.service });
+
+    this.order.set({}, { validate: true });
 
-    $.post({
-      url: app.getServerUrl('ob/purchase'),
-      data: JSON.stringify(this.order.toJSON()),
-    })
-      .done((data) => {
-        console.log(data);
+    if (!this.order.validationError) {
+      $.post({
+        url: app.getServerUrl('ob/purchase'),
+        data: JSON.stringify(this.order.toJSON()),
       })
-      .fail((data) => {
-        console.log(data);
+        .done((data) => {
+          console.log(data);
+        })
+        .fail((data) => {
+          console.log(data);
+        });
+    } else {
+      let errMsg = '';
+      _.each(this.order.validationError, (obj) => {
```

Comment:
Why use underscore utility functions when native JS has a well supported equivalent (in this case forEach, which is supported all the way back to IE9)?